libtcmu.c

```diff
@@ -536,8 +538,11 @@ static int add_device(struct tcmulib_context *ctx, char *dev_name,
 		tcmu_cfgfs_dev_exec_action(dev, "block_dev", 0);
 err_free:
 	free(dev);
-
 	return -ENOENT;
+err_nohandler:
+	free(dev);
+
+	return 1;
```

Comment:
In general, It makes more sense to return -1/1 for all those str***() function errors, and return ENOENT for nohandler case.